### ISSUE TITLE
Add external plugins to our input parser 

### ISSUE THREAD

**Issue by kingonly** [breez/breez-sdk-liquid#535]
There are legacy POS systems where partners have no control or ability to change their QRs. 
The idea here is to allow an SDK user to register external parser URLs that return valid lnurlp/bolt11 in response to a given string. 
One example is here: https://docs.google.com/document/d/1g_DAmaQY66uo1oXJNQutMnVw_sQjDhjJoeme2LOXets/edit?usp=sharing

We want our partners to be able to register https://staging.cryptoqr.net/.well-known/lnurlp/ and an external input parser. Input strings passed to the plugin should be url encoded.

**Comment by danielgranhao** [comment]
@kingonly Should we support the specific implementation by Moneybadger? IMO it's too specific. It entails a Lightning address-style lookup but is not compliant because it skips the 2nd request in LNURL-pay. 

We could keep our implementation generic by requiring partners to provide an input parser server that returns any format we support in the SDK (as would usually be provided to `parse()`). This would make it easier for others to use this feature and allow resolving to types other than bolt11.

The downside is that it requires the currently interested partner (MoneyBadger) to implement an additional endpoint.

**Comment by kingonly** [comment]
From a business standpoint, yes. We need to support MoneyBadger. 

Not sure I understand the problem. The parse return a valid lnurl-pay URL: https://breez-tech.slack.com/archives/C07SLM3H7D2/p1729678850977259

**Comment by danielgranhao** [comment]
Indeed, it does. The Google doc isn't up to date. Thanks

**Comment by roeierez** [comment]
@danielgranhao I think you are right about trying to give here a generic solution as long as it is not adding too much of a complexity.
I was thinking about letting the user define a list of input handler configurations that each of them takes:
* regex to qualify the input
* URL with a place holder (for the input)  to invoke as an external endpoint which resolves the lnurlpay or anything our input 
parser supports (can actually be arbitrary input: bolt11, offer etc..)
* We can also add name of the external provider

We can use these handlers on unidentified input untill one of them resolves.
What do you think?

**Comment by danielgranhao** [comment]
@roeierez I like it. But again, that means MoneyBadger would need to implement a compatible parser. If that's acceptable, I'll proceed with a proposal for the implementation.

**Comment by roeierez** [comment]
Right! As far as I understand they have that endpoint implemented and the SDK responsibility is only to invoke that external URL (parser).

**Comment by danielgranhao** [comment]
They provide an LNURL-pay server. The response to a request to their endpoint is something like: 
```
{"callback":"<callback_url>","minSendable":57000,"maxSendable":57000,"metadata":"[[\"text/plain\", \"Pick n Pay Payment\"]]","tag":"payRequest"}
```
This isn't something we can work with directly using the SDK's parser. We would need to make a 2nd request to the callback to get something useful (the payment request).

This can work, of course, but it doesn't make for a generic solution.

**Comment by kingonly** [comment]
I don't understand. All we need to do is to add a regex to parse the response from the parser. This can be done generically. Meaning the engine is generic, the configuration is specific. 

We can the parser url with the QR data (configuration), we parse the response to get the lnurlp end point and from there it's just our regular flow. What am I missing? 

https://docs.google.com/document/d/1g_DAmaQY66uo1oXJNQutMnVw_sQjDhjJoeme2LOXets/edit?usp=drivesdk

**Comment by danielgranhao** [comment]
@roeierez and I continued the discussion out-of-band, and I'm now working on a solution that doesn't require any changes from MoneyBadger's side. 

The point was that Moneybadger's parser doesn't return an lnurlp endpoint. The parser **is** the lnurlp endpoint itself.
I'm nonetheless working on a solution that should work anyway.

**Comment by kingonly** [comment]
Nope, the parser isn't the lnurlp endpoint. That's the reason their service exists. It's to convert legacy POS QRs to lnurlp. See the document and the discussion in the MoneyBadger channel.

**Comment by roeierez** [comment]
@kingonly I thought the same but it turns out the end point combined with the input data is the url for lnurl pay. Invoking this url returns the first response of lnurl protocol and not the lnurl itself.
We can handle that by returning this constructed url as the lnurl pay to the user which meets our use case requirements

**Comment by kingonly** [comment]
All good.

**Comment by danielgranhao** [comment]
Closed by #597. Follow up: #599
